### PR TITLE
Add: files of js-movie-seat-booking task by andrewklmn

### PR DESCRIPTION
**Pull request by andrewklmn**
This is js-movie-seat-booking with an optional confirmation window (press 'Buy now' button after seat selection).

[Demo](https://andrewklmn.github.io/html-css-cinema-booking/dist/) | [Code](https://github.com/kottans/frontend-2021-classroom/tree/andrewklmn/js-movie-seat-booking/submissions/andrewklmn/js-movie-seat-booking)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  if (selectedDate !== lastSelected.date
+      || selectedTime !== lastSelected.time) {
+    selectedSeats.forEach(seat => { seat.checked = false; });
+    lastSelected.date = selectedDate;
+    lastSelected.time = selectedTime;
+    draftTicket.innerHTML = initialTicketView;
+    return;
+  }
+
+  let total = 0;
```
Comment (A-Ostrovnyy) [inline submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js]: Fix height of order table [Watch video](https://share.vidyard.com/watch/wAEoVhLF1j4NnwTw5c3TZq?). Also, fix reset order logic.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Sorry for the late responce.
Good work but one issue need to be fixed.

File: submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js [hunk 2]
```diff
@@ -11,6 +11,78 @@ import {
...
+    return sequence.join('-');
+  }).join(',');
+}
+
+function getSeatPosition(seatElement) {
+  const seatPosition = seatElement.id.split('_');
+  return {
+    row: seatPosition[1],
+    seat: seatPosition[2],
+  };
```
Comment (A-Ostrovnyy) [inline submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js]:
```suggestion
  const [row, seat] = seatElement.id.split('_');
  return {
    row,
    seat
  }
```
Don't forget about [Destructuring assignment](https://javascript.info/destructuring-assignment) )

File: submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js [hunk 3]
```diff
@@ -11,6 +11,78 @@ import {
...
+
+  let answer = '';
+
+  selectedRows.forEach((nextRow) => {
+    const seatsInRow = [];
+    selectedSeats.forEach((x) => {
+      const { row, seat } = getSeatPosition(x);
+      if (nextRow === row) {
+        seatsInRow.push(seat);
+      }
```
Comment (A-Ostrovnyy) [inline submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js:79]: forEach with push() in callback function you can replace with [].map(), but it's just my opinion, rewrite or not it's up to you)

File: submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js [hunk 4]
```diff
@@ -11,6 +11,78 @@ import {
 
 import getSelectedSeats from './seat-selector';
 
+const MAX_DETAILED_SEATS_IN_TICKET = 6;
+const MAX_LENGTH_OF_SIMPLE_SEAT_LIST = 4;
```
Comment (A-Ostrovnyy) [inline submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js:15]: 🔥

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Very creative solution)
Good work, couple of comments for improving the readability of your code.

File: submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js (hunk 3 above)
Comment (andrewklmn) [reply submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js:79]: I thought about [].map() here, but this callback pushes only a few seats from 'selectedSeats' (array with all rows and seats) when  IF condition is true...  
So the length of the result array can be less than the length of selectedSeats array...
This is why I selected forEach() with [].push() in callback instead.

Comment (A-Ostrovnyy) [reply submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js:79]: My bad I mean [].filter() )

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**